BUILD.bazel

```diff
@@ -0,0 +1,309 @@
+        "include/maliput/geometry_base/road_geometry.h",
+        "include/maliput/geometry_base/segment.h",
+        "include/maliput/geometry_base/strategy_base.h",
+    ],
+    copts = ["-std=c++17"],
+    strip_include_prefix = "include",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":api",
+    ],
+)
+
+cc_library(
+    name = "routing",
+    srcs = [
+        "src/routing/derive_lane_s_routes.cc",
+        "src/routing/find_lane_sequences.cc",
+        "src/routing/phase.cc",
+        "src/routing/route.cc",
+        "src/routing/routing_constraints.cc",
+    ],
+    hdrs = [
+        "include/maliput/routing/derive_lane_s_routes.h",
+        "include/maliput/routing/find_lane_sequences.h",
+        "include/maliput/routing/lane_s_range_relation.h",
+        "include/maliput/routing/phase.h",
+        "include/maliput/routing/route_position_result.h",
+        "include/maliput/routing/route.h",
+        "include/maliput/routing/router.h",
+        "include/maliput/routing/routing_constraints.h",
+    ],
+    copts = ["-std=c++17"],
+    strip_include_prefix = "include",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":api",
+        ":common",
+        ":geometry_base"
+    ],
+)
+
+cc_library(
+    name = "utility",
+    srcs = [
+        "src/utility/file_utils.cc",
+        "src/utility/generate_obj.cc",
+        "src/utility/generate_string.cc",
+        "src/utility/generate_urdf.cc",
+        "src/utility/mesh_simplification.cc",
+        "src/utility/segment_analysis.cc",
+        "src/utility/thread_pool.cc",
+    ],
+    hdrs = [
+        "include/maliput/utility/file_utils.h",
+        "include/maliput/utility/generate_obj.h",
+        "include/maliput/utility/generate_string.h",
+        "include/maliput/utility/generate_urdf.h",
+        "include/maliput/utility/mesh.h",
+        "include/maliput/utility/mesh_simplification.h",
+        "include/maliput/utility/segment_analysis.h",
+        "include/maliput/utility/thread_pool.h",
+    ],
+    copts = ["-std=c++17"],
+    strip_include_prefix = "include",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":api",
+        ":common",
+        ":math",
+        "@fmt",
+    ],
+)
+
+cc_library(
+    name = "plugin",
+    srcs = [
+        "src/plugin/create_road_network.cc",
+        "src/plugin/maliput_plugin_manager.cc",
+        "src/plugin/maliput_plugin.cc",
+    ],
+    hdrs = [
+        "include/maliput/plugin/create_road_network.h",
+        "include/maliput/plugin/maliput_plugin_architecture.h",
+        "include/maliput/plugin/maliput_plugin_manager.h",
+        "include/maliput/plugin/maliput_plugin_type.h",
+        "include/maliput/plugin/maliput_plugin.h",
+        "include/maliput/plugin/road_network_loader.h",
+    ],
+    copts = ["-std=c++17"],
+    linkopts = ['-ldl'],
+    strip_include_prefix = "include",
+    visibility = ["//visibility:public"],
+    deps = [
```

Comment:
Not super familiar with bazel but, here in the plugin library we link agaisnt `dl` for loading libraries in runtime, is it necessary to add it here as well?

See [cmake config](https://github.com/maliput/maliput/blob/e107e58fdd94f1d0aa07ff5dc3e4b004465a9994/src/plugin/CMakeLists.txt)

Reply:
I'm still learning too - setting ` linkopts = ['-ldl'],` at least gets it to build and run, but I don't know what I need to depend on (in a bazel-verse) or if in fact I need to do so.